### PR TITLE
Introduce a 'Create New Hyphae' special url and top bar highlight

### PR DESCRIPTION
**Pull request by roselandgoose**
- Adds a /new handler which simply redirects to a queried page.
- Prevents redirect on empty input by sending NO CONTENT
- Recognizes special case header link /new and creates input bar.

### CODE REVIEW

File: static/default.css [hunk 1]
```diff
@@ -656,6 +656,17 @@ kbd {
 	display: block;
 	padding: 0 .5rem;
 }
+.top-bar__highlight-new {
+    display: block;
```
Comment (handlerug) [inline static/default.css:660]: Nitpick: indentation

File: web/stuff.go [hunk 2]
```diff
@@ -35,6 +35,14 @@ func initStuff(r *mux.Router) {
 	r.HandleFunc("/favicon.ico", func(w http.ResponseWriter, rq *http.Request) {
 		http.Redirect(w, rq, "/static/favicon.ico", http.StatusSeeOther)
 	})
+	r.HandleFunc("/new", func(w http.ResponseWriter, rq *http.Request) {
+		new_path := rq.URL.Query().Get("new-path")
```
Comment (handlerug) [inline web/stuff.go:39]: Nitpick: prefer camelCase for variable names

**Review by handlerug — CHANGES_REQUESTED** [review]
Thanks for the contribution! Left some nitpicks during the review, but otherwise looks very good to me. Here's how it looks in Chrome:
image

And the same thing but if I add it to the end:
image

The "replacing /new links with an input field" idea seems clever to me! Not sure if it fits well in the top bar, though; it looks messy, especially with those dark heavy outlines (you _could_ blame Chrome for that, but there were styles that removed it before). Maybe @bouncepaw can think of a better placement for it? He also had an idea of a big "New" button for creating hyphae, and I had an idea of an ultimate JavaScript-enhanced field with existing hyphae autocompletion and fast editing shortcuts.

I'd also love some way of creating new hyphae relative to the one the user is currently at. So, if the user is currently browsing Fruits/Apple and they want to add a subhypha, they can type in "./Nutrients" and have Fruits/Apple/Nutrients hypha opened for editing. You just need to resolve the new path relative to the URL the user is coming from (`Referer` HTTP header), like it's done for links in Mycomarkup.

File: views/stuff.qtpl [hunk 3]
```diff
@@ -58,7 +58,13 @@
 					<li class="top-bar__section top-bar__section_highlights">
 						<ul class="top-bar__highlights">
 						{%- for _, link := range cfg.HeaderLinks -%}
-						{% if link.Href != "/" %}
+						{% if link.Href == "/new" %}
+							<li class="top-bar__highlight">
+								<form class="top-bar__highlight-new" method="GET" action="/new">
+									<input class="top-bar__highlight-new-input" size="{%s inputSize(link.Display) %}" type="text" name="new-path" placeholder="{%s link.Display %}">
```
Comment (bouncepaw) [inline views/stuff.qtpl:64]: Few things.

* A second input line in the top bar is not a good idea.
* This input offers nothing more than the address bar in any browser.

**Review by bouncepaw — COMMENTED** [review]
Although I like the whole idea of having a way to create _a_ hypha, this pull request does not solve the problem.

The proposed input line in the top bar plays the role of a stripped-down address bar in the browser. @handlerug had an idea of relative names in that input; it is already supported by the address bar.

What would work is a link to a page where you create a hypha. A link, not an input, to not force users to think of the name beforehand.

**Comment by bouncepaw** [comment]
I'm closing this pull request for now because it does not solve the problem and offers nothing new.

Nonetheless, we sure need something like that! I will open an issue regarding that. Thanks for bringing the attention to the topic.

**Pull request closed without merging**